src/Aspire.Hosting.PostgreSQL/PostgresServerResource.cs

```diff
@@ -88,6 +88,20 @@ public ReferenceExpression ConnectionStringExpression
     /// </summary>
     public IReadOnlyDictionary<string, string> Databases => _databases;
 
+    internal void SetPassword(ParameterResource password)
+    {
+        ArgumentNullException.ThrowIfNull(password);
+
+        PasswordParameter = password;
```

Comment:
These properties are publically settable. We can remove these methods and just set the property.